content/js1/sprints/4/prep/index.md

```diff
@@ -8,4 +8,666 @@ backlog= 'Module-JS1'
+`console.log` and `Math.round` are **API**s.
+
+{{</note >}}
+
+{{<note type="exercise" title="exercise">}}
+
+Explain why `Math.round` is an **API**
+
+{{</note>}}
+
+{{<note type="exercise" title="exercise">}}
+
+### 🔍 Figure it out
+
+There should be one more change to your project after running the installation command. Use documentation or ChatGPT to figure out what this change is for.
+{{</note>}}
+
+### 🖌️ First test case
+
+> 🎯 Goal: Write the a test for the case below, using Jest:
+
+- Case 1 💼
+
+```js
+const input = 1;
+const currentOutput = getOrdinalNumber(input);
+const targetOutput = "1st";
+```
+
+To work out how to write a test using Jest, we can check out the documentation.
+
+We can create a file called `get-ordinal-number.js` and in there write our first test.
+We can figure out how to write this test using the Jest documentation. We can start writing a test:
+
+get-ordinal-number.test.js
+
+```js
+test("converts 1 to an ordinal number", function () {});
+```
+
+Let's break down this syntax.
+The `test` function is part of the Jest API: it is an interface we use to perform a particular task.
+In particular, we're using `test` to create a test case.
+
+Let's break down the arguments we're passing to `test`:
+
+- 1st argument: `"converts 1 to an ordinal number"`, which describes the behaviour we're testing for
+- 2nd argument: `function () {}`, we will write some assertions in this `function () {}` to check the behaviour
+
+### ⚖️ Creating assertions
+
+Now we have
+
+get-ordinal-number.test.js
+
+```js
+test("converts 1 to an ordinal number", function () {});
+```
+
+We need to write an **assertion** inside the body of `function () {}`
+
+> 🗣️ Recall: The assertion is the part of the test code that actually checks to see if something is true or not.
+
+In this example, we want to check that the following is true:
+
+We expect `getOrdinalNumber(1)` to be `"1st"`
+
+An assertion in Jest looks like this:
+
+```js
+expect(getOrdinalNumber(1)).toBe("1st");
+```
+
+The function `toBe` is used to check that the current output of `getOrdinalNumber(1)` and the target output of `"1st"` are equal to each other.
+`toBe` is just one example of a function called a [matcher](https://jestjs.io/docs/using-matchers).
+A matcher is a function we use to compare values in Jest.
+
+So we can combine this with the test we wrote earlier, to get our first complete test case:
+
+```js
+test("converts 1 to an ordinal number", function () {
+  expect(getOrdinalNumber(1)).toBe("1st");
+});
+```
+
+### 👟 Running tests
+
+We can try running the file index.test.js with node in the following way:
+
+```bash
+node index.test.js
+```
+
+but we get an error:
+
+```bash
+ReferenceError: test is not defined
+```
+
+This is because `test` isn’t defined anywhere in the file.
+
+So we need to execute this file so that the Jest API is available in our file. We can do this by adding executing the test file using Jest: we do this using an npm script.
```

Comment:
```suggestion
So we need to execute this file so that the Jest API is available in our file. We can do this by executing the test file using Jest: we do this using an npm script.
```